### ISSUE TITLE
test overhead fails

### ISSUE THREAD

**Issue by edhartnett** [NOAA-GSD/GPTL-fortran#4]
When I run the fortran test overhead, I get this output:

```
ed@mikado:~/noaa_tmp/GPTL-fortran/test$ ./overhead 
 overhead: running gettimeofday nthreads=           1
 overhead: running nanotime nthreads=           1
GPTL error:GPTL: init_nanotime: not enabled
```

What's up with that?

**Comment by jmrosinski** [comment]
I'm getting a different error when I run "make check". But, the symptom of
your error I think is that HAVE_NANOTIME in the C build is "no". So you're
probably encountering the issue of the "overhead" code normally being run
manually rather than part of "make check". So to fix it if it's to be part
of "make check" will require an ifdef HAVE_NANOTIME. But I'm OK with taking
this code out of the automated tests if you'd prefer.

This is something which in my stuff the user has to set manually. It would
be great to automate it since most users won't know if it's available or
not. If you look in gptl.c, you'll see inside of #ifdef HAVE_NANOTIME,
assembler code:

asm volatile ("rdtsc":"=a" (a), "=d" (d));

and

__asm__ __volatile__("rdtsc":"=A" (val): );

Perhaps an automated test could check if this succeeds or not?

If this is too involved, I'm OK with something like --enable-nanotime arg
to configure.

The HAVE_NANOTIME capability is important. Reading from a register is
orders of magnitude faster and more accurate than gettimeofday()

**Comment by edhartnett** [comment]
First I will put in an --enable-nanotime in both C and Fortran builds.

Of course it would be much better to test. Can you provide me a c program which will compile if nanotime is present, and fail to compile if it is absent? That would be ideal...

**Comment by jmrosinski** [comment]
Yes I'll have a look at that. Not sure if we also need a runtime
verification that it not only compiles but behaves correctly. Remember it's
an inserted assembly instruction. As I recall PGI can't handle it. Also, it
only works on Intel/AMD hardware.

**Comment by jmrosinski** [comment]
(note that this issue affects both C and Fortran equally) Sorry for the
delay, I've got a code that so far works everywhere. It'd be nice to find a
compiler on which it fails. gcc, nvcc, pgcc, and MacOS cc all succeed. Feel
free to install anyway but if you have access to e.g. IBM C compiler that'd
be a good one to check. Here's the code, which can be simplified to remove
the printf and #include.

#include <stdio.h>
int main ()
{
  unsigned int a = 0, d = 0;
  asm volatile ("rdtsc":"=a" (a), "=d" (d));
  printf ("a=%u d=%u\n", a, d);
}

**Comment by edhartnett** [comment]
OK, now I check for x86 and set HAVE_NANOTIME successfully. But now test overhead gets this error:


```
ed@mikado:~/noaa_tmp/GPTL-fortran/test$ ./overhead
 overhead: running gettimeofday nthreads=           1
 overhead: running nanotime nthreads=           1
GPTL error:GPTL: init_nanotime: not enabled
```

Is this because I have to do the same check for HAVE_NANOTIME in the C library? I will try and see...

**Comment by edhartnett** [comment]
OK, I made the nanotime modification to the C library, and tried again, but still I get this error:

```
ed@mikado:~/noaa_tmp/GPTL-fortran/test$ ./overhead 
 overhead: running gettimeofday nthreads=           1
 overhead: running nanotime nthreads=           1
GPTL error:GPTL: init_nanotime: not enabled
```

I am going to put my work so far up as a PR while we continue to figure out what is wrong with this test.

**Comment by edhartnett** [comment]
@jmrosinski can you build the latest versions of both C and Fortran and see if you get this error too? Or do you still get some other error?
